js-pkg/client/src/provider.ts

```diff
@@ -357,35 +405,19 @@ export class YSweetProvider {
   }
 
   private websocketError(event: Event) {
-    this.emit('connection-error', event)
+    this.emit(EVENT_CONNECTION_ERROR, event)
+    this.setStatus(STATUS_ERROR)
 
-    if (this.shouldConnect && !this.isConnecting) {
-      this.connect()
-    }
+    this.connect()
   }
 
-  protected emit(eventName: YSweetEvent, data: any = null): void {
+  public emit(eventName: YSweetEvent | YWebsocketEvent, data: any = null): void {
```

Comment:
Why make this public?

Reply:
It's needed so that `WebSocketCompatLayer` can emit its own events, unfortunately.